src/main/webapp/app/shared/statistics-graph/statistics-average-score-graph.component.ts

```diff
@@ -153,4 +178,173 @@ export class StatisticsAverageScoreGraphComponent implements OnInit {
+        this.currentPeriod = 0;
+        this.setupChart(this.currentlyDisplayableExercises);
+        this.currentSize = this.currentlyDisplayableExercises.length;
+    }
+
+    /**
+     * Performs a filtering for a performance interval (legend entry).
+     * If all intervals are currently displayed and one is clicked, for this interval is filtered.
+     * If a disabled interval is clicked, this interval is added to the filter.
+     * @param interval the interval that is selected
+     */
+    togglePerformanceInterval(interval: PerformanceInterval): void {
+        const currentValue = this.displayColorMap.get(interval);
+        // we reset the view of the chart (changed with the arrows) before displaying the filtered exercises
+        this.currentPeriod = 0;
+        let newValue = '';
+        // This means an interval is selected that is currently already displayed
+        if (currentValue !== '') {
+            // if only this interval is selected, reselecting it leads to selecting all intervals
+            if (this.numberOfSelectedIntervals === 1) {
+                this.includeAllIntervals();
+                this.exerciseScoresFilteredByPerformanceInterval = this.orderAverageScores(this.exerciseAverageScores);
+            } else {
+                this.deselectAllOtherIntervals(interval);
+            }
+            this.initializeFilterOptionsAndSetupChartWithCurrentVisibleScores();
+            return;
+        }
+        switch (interval) {
+            case PerformanceInterval.LOWEST: {
+                newValue = this.CRITICAL_CLASS;
+                break;
+            }
+            case PerformanceInterval.AVERAGE: {
+                newValue = this.MEDIAN_CLASS;
+                break;
+            }
+            case PerformanceInterval.BEST: {
+                newValue = this.BEST_CLASS;
+                break;
+            }
+        }
+        // This map determines whether the color legend next to a chart entry is colored or not representing whether this entry is currently visible in the chart or not
+        this.displayColorMap.set(interval, newValue);
+        this.numberOfSelectedIntervals += 1;
+        const exercises = this.filterForPerformanceInterval(interval);
+        this.exerciseScoresFilteredByPerformanceInterval = this.orderAverageScores(this.currentlyDisplayableExercises.concat(exercises));
+        this.initializeFilterOptionsAndSetupChartWithCurrentVisibleScores();
+    }
+
+    /**
+     * Auxiliary method that identifies the exercises contained by the passed interval
+     * @param interval the interval the exercises should be filtered against
+     * @private
+     */
+    private filterForPerformanceInterval(interval: PerformanceInterval) {
+        let filterFunction;
+        switch (interval) {
+            case PerformanceInterval.LOWEST: {
+                filterFunction = (model: CourseManagementStatisticsModel) => model.averageScore <= this.weakestThirdUpperBoundary;
```

Comment:
I assume that this may be related to the fact, that this course only contains one exercise. When there is only one exercise, the average score of all exercises equals the average score for this particular exercise. As you also allow equality ('=') in the cases for both LOWEST and BEST when comparing the averages, in both cases the exercise will be shown when filtering for lowest or best third interval.

Reply:
I changed the implementation in that way, that the thresholds are now exclusive. I think this should fix the problem.
![image](https://user-images.githubusercontent.com/80622272/155127282-1a84a592-6eb2-43a1-9e1f-7888bd2c6c78.png)

Reply:
Works as described, thanks.